bin/dustc

```diff
@@ -36,47 +38,74 @@ if (!cli.argc) {
-.reduce(function(a, b) { return a.concat(b); }, [])
-.forEach(function(inputFile, index, filesToProcess) {
-    read(inputFile, function(err, data) {
-        if (err) {
-            cli.info('Couldn\'t open ' + inputFile + ' for reading');
-            return;
-        }
-
-        var outputFile = cli.options.output,
-            templateName = path.join(path.dirname(inputFile),
-                                     path.basename(inputFile, path.extname(inputFile))),
-            compiledData;
-
-        // Use the template's path as the output path if split-files is turned on
-        if (cli.options.split) {
-            outputFile = templateName + '.js';
-        }
-
-        // Allow override of template name as long as there's only one template
-        if (cli.options.name && filesToProcess.length === 1) {
-            templateName = cli.options.name;
-        }
+function handle() {
+    streams = {};
+    paths.forEach(function(inputFile, index, filesToProcess) {
+        read(inputFile, function(err, data) {
+            if (err) {
+                cli.info('Couldn\'t open ' + inputFile + ' for reading');
+                return;
+            }
+
+            var outputFile = cli.options.output,
+                templateName = path.join(path.dirname(inputFile),
+                                         path.basename(inputFile, path.extname(inputFile))),
+                compiledData;
+
+            // Use the template's path as the output path if split-files is turned on
+            if (cli.options.split) {
+                outputFile = templateName + '.js';
+            }
+
+            // Allow override of template name as long as there's only one template
+            if (cli.options.name && filesToProcess.length === 1) {
+                templateName = cli.options.name;
+            }
+
+            // Optionally strip leading directories from a template name
+            // For example, if --pwd=tmpl, `tmpl/foo/a` becomes `foo/a`
+            if (cli.options.pwd) {
+                templateName = path.relative(cli.options.pwd, templateName);
+            }
+
+            compiledData = compile(data, templateName);
+            output(compiledData, outputFile);
+        });
+    });
+}
 
-        // Optionally strip leading directories from a template name
-        // For example, if --pwd=tmpl, `tmpl/foo/a` becomes `foo/a`
-        if (cli.options.pwd) {
-            templateName = path.relative(cli.options.pwd, templateName);
+/**
+ * Turn on watching if --watch is enabled
+ * Watching runs the compilation again if any file changes or is deleted.
+ * However, adding new files means, for now, that you need to restart the watch.
+ */
+if(cli.options.watch) {
+    cli.info('Watching for changes... ^C to quit');
+    watchr.watch({
+        paths: paths,
+        listeners: {
+            change: function(method, file) {
+                cli.info(method + ' ' + file);
+                handle();
```

Comment:
Everything, because you want to regenerate the entire file in the case that you're redirecting output to a single location.